src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/maven/WriteMavenPomStepTest.java

```diff
@@ -64,20 +64,20 @@ public void setup() throws Exception {
     public void testWriteAndRead() throws Exception {
         WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
         p.setDefinition(new CpsFlowDefinition(
-                "node('slaves') {\n" +
-                        "  dir('inhere') {\n" +
-                        "    Model pom = new Model()\n" + //checks the auto import
-                        "    pom.artifactId = 'my-test-project'\n" +
-                        "    pom.groupId = 'com.example.jenkins.test'\n" +
-                        "    pom.version = '1.1-SNAPSHOT'\n" +
-                        "    Dependency d = new Dependency()\n" +
-                        "    d.artifactId = 'pipeline-utility-steps'\n" +
-                        "    d.groupId = 'org.jenkins-ci.plugins'\n" +
-                        "    d.version = '1.0'\n" +
-                        "    d.classifier = 'hpi'\n" +
-                        "    pom.addDependency(d)\n" +
-                        "    writeMavenPom(pom)\n" +
-                        "  }\n" +
+                "def doWrite() {\n" +
```

Comment:
This needs to be `@NonCPS`.

Reply:
I originally made them NonCPS, but then I reverted when it worked in any way. The context XMLs are likely weird, but 🤷‍♂️